src/kaleidoscope/hardware/olkb/Planck.h

```diff
@@ -22,28 +22,26 @@
 #include <Arduino.h>
 #define HARDWARE_IMPLEMENTATION kaleidoscope::hardware::olkb::Planck
 
-#include "kaleidoscope/hardware/olkb/planck/PlanckScannerDescription.h"
+#include "kaleidoscope/hardware/olkb/planck/PlanckKeyScannerDescription.h"
 
 #include "kaleidoscope/driver/led/NoLeds.h"
-#include "kaleidoscope/driver/mcu/AVR.h"
-#include "kaleidoscope/driver/scanner/ATMegaScanner.h"
+#include "kaleidoscope/driver/mcu/ATMega32U4.h"
+#include "kaleidoscope/driver/keyscanner/ATMegaKeyScanner.h"
 
-#include "kaleidoscope/HardwareDescription.h"
-#include "kaleidoscope/Hardware.h"
+#include "kaleidoscope/DeviceDescription.h"
+#include "kaleidoscope/Device.h"
 
 namespace kaleidoscope {
 namespace hardware {
 namespace olkb {
 
-using PlanckScanner = kaleidoscope::driver::scanner::ATMegaScanner<PlanckScannerDescription>;
-
-struct PlanckDescription : public kaleidoscope::HardwareDescription {
-  typedef PlanckScannerDescription ScannerDescription;
-  typedef PlanckScanner KeyScanner;
-  typedef kaleidoscope::driver::mcu::AVR MCU;
+struct PlanckDeviceDescription : kaleidoscope::DeviceDescription {
```

Comment:
Not sure if `DeviceDescription` might better be named `NoOpDeviceDescription` to be in sync with the names of the noop controllers.

Reply:
Or something like `RootDeviceDescription`, `BaseDeviceDescription`, etc. Something that explains that `DeviceDescription` is not a final class that is meant to be used as is.

Reply:
`DeviceDescriptionRoot`, or `DeviceDescriptionBase`? Can't explain well _why_, but putting `Base`/`Root` at the end _feels_ better to me.

Reply:
But the why `PlanckScanner` and not `ScannerPlanck`? AFAIK in OOP it is most common to use prefix to express specialisation. But...you know what...I don't care :smirk:

Reply:
After trying `BaseDeviceDescription`, `DeviceDescriptionBase`, and the `Root` variants, and the `NoOp` ones too, I came to the conclusion that for `Device` and `DeviceDescription`, I'd stick with the current names. They _feel_ better. Without the base/root/noop designation, it still reads like a generic device(description) we're deriving from, which is fine.

On the other hand, I'll be adding a base class for the noop controllers, and the noop ones will be simple, no-op derivatives. That should streamline the naming, I think.